sympy/combinatorics/perm_groups.py

```diff
@@ -5075,3 +5075,132 @@ def _stabilizer(degree, generators, alpha):
     return [_af_new(x) for x in stab_gens]
 
 PermGroup = PermutationGroup
+
+class SymmetricPermutationGroup(Basic):
+    """
+    The class defining the lazy form of SymmetricGroup
+
+    deg : int
+    """
+
+    def __new__(cls, deg):
+        deg = sympify(deg)
+        obj = Basic.__new__(cls, deg)
+        obj.deg = deg
+        obj._order = None
+        return obj
+
+    def order(self):
+        """
+        Return the order of the permutation group.
+        """
+        if self._order is not None:
+            return self._order
+        n = self.deg
+        self._order = factorial(n)
+        return self._order
+
+    @property
+    def degree(self):
+        """
+        Return the degree of the permutation group.
+        """
+        return self.deg
+
+    @property
+    def identity(self):
+        '''
+        Return the identity element of the permutation group.
+
+        '''
+        return _af_new(list(range(self.deg)))
+
+class Coset(Basic):
+    """A left coset of a permutation group with respect to an element.
+
+    Parameters
+    ==========
+
+    g : Permutation
+
+    H : PermutationGroup
+
+    dir : "+" or "-", If not specified by default it will be "+"
+        here ``dir`` specified the type of coset "+" represent the
+        right coset and "-" represent the left coset.
+
+    G : PermutationGroup, optional
+        The group which contains *H* as its subgroup and *g* as its
+        element.
+
+        If not specified, it would automatically become a symmetric
+        group ``SymmetricPermutationGroup(g.size)`` and
+        ``SymmetricPermutationGroup(H.degree)`` if ``g.size`` and ``H.degree``
+        are matching.``SymmetricPermutationGroup`` is a lazy form of SymmetricGroup
+        used for representation purpose.
+    """
+
+    def __new__(cls, g, H, G=None, dir = "+"):
+        g = sympify(g)
```

Comment:
You should use _sympify because we have decided to use https://github.com/sympy/sympy/issues/18066#issue-539262305